test/workers/test_websocket_worker.py

```diff
@@ -81,166 +64,142 @@ def create_self_signed_cert(cert_path, key_path):
+    server.terminate()
+
 
+def test_list_objects_remote(hook, start_remote_worker):
+    server, remote_proxy = start_remote_worker(id="fed-list-objects", hook=hook)
+    remote_proxy.clear_objects()
 
-def test_list_objects_remote(hook, start_proc):
+    x = torch.tensor([1, 2, 3]).send(remote_proxy)
 
-    kwargs = {"id": "fed", "host": "localhost", "port": 8765, "hook": hook}
-    process_remote_fed1 = start_proc(WebsocketServerWorker, **kwargs)
-    local_worker = instantiate_websocket_client_worker(**kwargs)
+    res = remote_proxy.list_objects_remote()
 
-    x = torch.tensor([1, 2, 3]).send(local_worker)
+    x = torch.tensor([1, 2, 3]).send(remote_proxy)
 
-    res = local_worker.list_objects_remote()
+    res = remote_proxy.list_objects_remote()
     res_dict = eval(res.replace("tensor", "torch.tensor"))
     assert len(res_dict) == 1
 
-    y = torch.tensor([4, 5, 6]).send(local_worker)
-    res = local_worker.list_objects_remote()
+    y = torch.tensor([4, 5, 6]).send(remote_proxy)
+    res = remote_proxy.list_objects_remote()
     res_dict = eval(res.replace("tensor", "torch.tensor"))
     assert len(res_dict) == 2
 
-    # retrieve x and y before terminating the websocket connection
-    x.get()
-    y.get()
-    local_worker.close()
-    local_worker.remove_worker_from_local_worker_registry()
-    process_remote_fed1.terminate()
-
+    # delete x before terminating the websocket connection
+    del x
+    del y
+    time.sleep(0.1)
+    remote_proxy.close()
+    time.sleep(0.1)
+    remote_proxy.remove_worker_from_local_worker_registry()
+    server.terminate()
 
-def test_objects_count_remote(hook, start_proc):
 
-    kwargs = {"id": "fed", "host": "localhost", "port": 8764, "hook": hook}
-    process_remote_worker = start_proc(WebsocketServerWorker, **kwargs)
-    local_worker = instantiate_websocket_client_worker(**kwargs)
+def test_objects_count_remote(hook, start_remote_worker):
+    server, remote_proxy = start_remote_worker(id="fed-count-objects", hook=hook)
+    remote_proxy.clear_objects()
 
-    x = torch.tensor([1, 2, 3]).send(local_worker)
+    x = torch.tensor([1, 2, 3]).send(remote_proxy)
 
-    nr_objects = local_worker.objects_count_remote()
+    nr_objects = remote_proxy.objects_count_remote()
     assert nr_objects == 1
 
-    y = torch.tensor([4, 5, 6]).send(local_worker)
-    nr_objects = local_worker.objects_count_remote()
+    y = torch.tensor([4, 5, 6]).send(remote_proxy)
+    nr_objects = remote_proxy.objects_count_remote()
     assert nr_objects == 2
 
     x.get()
-    nr_objects = local_worker.objects_count_remote()
+    nr_objects = remote_proxy.objects_count_remote()
     assert nr_objects == 1
 
-    # get remote object before terminating the websocket connection
-    y.get()
-    local_worker.close()
-    local_worker.remove_worker_from_local_worker_registry()
-    process_remote_worker.terminate()
-
-
-def test_clear_objects_remote(hook, start_proc):
```

Comment:
why did you remove this test?

Reply:
good catch! that was me failing to merge the two files, will add it back.